### PR TITLE
HR-99-Beautify-the-UI

### PR DESCRIPTION
**Pull request by Viccsasa**
- Added the first level Tool Bar and a new component to add tools.
- Changed some icons and css styles of elements.
- Added a new look of color picker, size picker and eraser selector.

### CODE REVIEW

File: src/routes/canvas/toolbar/DrawingManager.tsx [hunk 1]
```diff
@@ -98,12 +102,17 @@ class DrawingManager extends React.Component<{inkingManager: InkingManager}> {
...
         return (
-            <div>
-                <Pen {...toolProps} />
-                <Highlighter {...toolProps} />
-                <Eraser {...toolProps} />
-                <LaserPointer {...toolProps} />
+            <div className='tool-second-level' style={{display: this.props.display}}>
+                {/* <FluentProvider theme={webLightTheme}> */}
+                <Toolbar className='popover'>
+                    <Pen {...toolProps} icon={ <img src={require("../../../assets/ink-pen.png")} alt="Icon" />}/>
```
Comment (Mike-Val) [inline src/routes/canvas/toolbar/DrawingManager.tsx]: The requiring of the assets could be moved to the beginning of the files after the imports. Or even better each tool could require their icon directly in their own file.

File: src/routes/canvas/toolbar/tools/Eraser.tsx [hunk 2]
```diff
@@ -1,15 +1,16 @@
 import React from "react";
 import { InkingTool } from "@microsoft/live-share-canvas";
 import { Button } from "@fluentui/react-components";
+import { Eraser24Regular, EraserSegment24Regular} from "@fluentui/react-icons"
 
 import Tool, { ToolProps } from "./Tool";
-import SizePicker from "./SizePicker";
+import MySizePicker from "./MySizePicker";
```
Comment (Mike-Val) [inline src/routes/canvas/toolbar/tools/Eraser.tsx]: Not a big fan of the name. If a more appropriate name can be used would be good, but it's just a minor thing

File: src/routes/canvas/toolbar/tools/Eraser.tsx [hunk 3]
```diff
@@ -73,14 +77,27 @@ class Eraser extends Tool {
...
-                        {this.state.isPointEraser ? eraserProps.icon : pointEraserProps.icon}
-                    </Button>
-                </div>}
-                {isDoubleClick && <SizePicker setSize={this.setSize} />}
+            <Tool {...this.props} icon={<img src={require("../../../../assets/eraser.png")} alt="Icon" />} tool={eraser.tool}>
+                <div className="tool-third-level">
+                    {/* Draw the button to select the other eraser */}
+                    {isDoubleClick &&
+                        <div>
+                            <Button onClick={() => this.setEraser(false)}
```
Comment (Mike-Val) [inline src/routes/canvas/toolbar/tools/Eraser.tsx]: Here the two buttons are defined explicitly. It is a bit of code duplication that could be avoided given that the only difference is the icon used.
Consider refactoring

File: src/routes/canvas/toolbar/tools/MyColorPicker.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+export interface MyColorPickerProps {
+  color: IColor;
+}
+
+/**
+ * A custom color picker component.
+ * This component uses Fluent UI's ColorPicker to allow the user to pick a color.
+ * The picked color is then passed up to the parent component through the setColor prop.
+ */
+class MyColorPicker extends React.Component<{defaultColor: string, setColor: (color: string) => void}, MyColorPickerProps> {
```
Comment (Mike-Val) [inline src/routes/canvas/toolbar/tools/MyColorPicker.tsx]: Might want to move the `defaultColor` and `setColor` props inside the `MyColorPickerPros` interface

File: src/routes/canvas/toolbar/tools/Eraser.tsx [hunk 5]
```diff
@@ -35,13 +36,15 @@ const pointEraserProps = {
 class Eraser extends Tool {
     static defaultProps = eraserProps;
     state = {
+        size: 5,
```
Comment (Mike-Val) [inline src/routes/canvas/toolbar/tools/Eraser.tsx]: After some tests, it seems that for the point eraser, a default value of 10 would work better.

**Review by Mike-Val — CHANGES_REQUESTED** [review]
Amazing job, I really love the results!!!

I only have 2 more comments that are related to the UX.

## Inconsistent look
Could we make the look of these two bars the same? 
Either style looks good; just use the same for both.
Screenshot 2023-07-24 at 16 34 30

## Improved experience
After selecting an option like the color for a tool, I would like the option picker to disappear when I start writing on the canvas.
An example of the current (wrong) behavior is:
https://github.com/Holo-Repository/MSTeams-WebApp/assets/39745976/8ee78f3b-5092-4eed-a512-cee7ca7f14bb

Addidionally, option menus should be closed when focusing another view, not just hidden.
An example of the wrong behavious is:
https://github.com/Holo-Repository/MSTeams-WebApp/assets/39745976/4eb5bfef-db6d-40b4-bec0-ff09a6cbe938
I would prefer if when clicking back on the drawing menu the already open color options would have instead been closed and only the last used tool would be remembered.


Let me know if my request are reasonable and if they make sense.
I also left some other minor suggestions in the code so check them out as well.
Regardless, amazing work nonetheless 👍🏻

**Comment by Viccsasa** [comment]
> Amazing job, I really love the results!!!
> 
> I only have 2 more comments that are related to the UX.
> 
> ## Inconsistent look
> Could we make the look of these two bars the same? Either style looks good; just use the same for both. Screenshot 2023-07-24 at 16 34 30
> 
> ## Improved experience
> After selecting an option like the color for a tool, I would like the option picker to disappear when I start writing on the canvas. An example of the current (wrong) behavior is: https://github.com/Holo-Repository/MSTeams-WebApp/assets/39745976/8ee78f3b-5092-4eed-a512-cee7ca7f14bb
> 
> Addidionally, option menus should be closed when focusing another view, not just hidden. An example of the wrong behavious is: https://github.com/Holo-Repository/MSTeams-WebApp/assets/39745976/4eb5bfef-db6d-40b4-bec0-ff09a6cbe938 I would prefer if when clicking back on the drawing menu the already open color options would have instead been closed and only the last used tool would be remembered.
> 
> Let me know if my request are reasonable and if they make sense. I also left some other minor suggestions in the code so check them out as well. Regardless, amazing work nonetheless 👍🏻

About the inconsistent look, I will change them to the same style and see how it looks like. The suggestions related to the interactive experience are great, will try to figure it out.

**Comment by Mike-Val** [comment]
Amazing!

I am merging https://github.com/Holo-Repository/MSTeams-WebApp/pull/5 so keep in mind that before merging your PR there will be some manual merging needed.

File: src/routes/canvas/toolbar/tools/Eraser.tsx (hunk 2 above)
Comment (Viccsasa) [reply src/routes/canvas/toolbar/tools/Eraser.tsx]: How about  SizeSelector / SizeSlider?

Comment (Mike-Val) [reply src/routes/canvas/toolbar/tools/Eraser.tsx]: Sure, sounds good!

**Review by Mike-Val — DISMISSED** [review]
I see that all the comments I left have been addressed, and I am very satisfied with the result!!

Now: TO THE MERGE!!

Pay attention that no functionality is lost when merging! If you require help feel free to reach out!!

**Comment by Mike-Val** [comment]
Force pushed to fix commit names (plus added some small refactorings)

**Review by Mike-Val — APPROVED** [review]
The code is satisfactory, and from Huanran's tests the functionality appears to have been merged correctly.

**Pull request merged**